lldb/source/Plugins/ObjectFile/Minidump/MinidumpFileBuilder.cpp

```diff
@@ -655,9 +655,10 @@ MinidumpFileBuilder::AddMemoryList(const lldb::ProcessSP &process_sp,
     const addr_t addr = core_range.range.start();
     const addr_t size = core_range.range.size();
     auto data_up = std::make_unique<DataBufferHeap>(size, 0);
+    Status read_error;
     const size_t bytes_read =
-        process_sp->ReadMemory(addr, data_up->GetBytes(), size, error);
-    if (bytes_read == 0)
+        process_sp->ReadMemory(addr, data_up->GetBytes(), size, read_error);
+    if (read_error.Fail() || bytes_read == 0)
```

Comment:
I lied: Someone might ask why this function isn't just using the LLVM ErrorOr<...> type, which forces you to handle errors, through dark C++ magic. You probably don't want to go changing the function signature, and then everywhere it's used, but someone might ask that.

Reply:
Nice, good to know